### PR TITLE
Fix NoSuchElementException in Transaction Receipt Logic

### LINKED ISSUE #2476: Fix gasPrice missing error on eth_getTransactionReceipt
```2021-06-24 15:05:42.548+00:00 | vert.x-worker-thread-19 | ERROR | JsonRpcHttpService | Error processing JSON-RPC requestBody
java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:148)
        at org.hyperledger.besu.ethereum.core.Transaction.calcEffectiveGas(Transaction.java:1025)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.TransactionReceiptResult.<init>(TransactionReceiptResult.java:77)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.TransactionReceiptStatusResult.<init>(TransactionReceiptStatusResult.java:26)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthGetTransactionReceipt.getResult(EthGetTransactionReceipt.java:57)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthGetTransactionReceipt.lambda$response$0(EthGetTransactionReceipt.java:48)
        at java.base/java.util.Optional.map(Optional.java:265)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthGetTransactionReceipt.response(EthGetTransactionReceipt.java:48)
        at org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcHttpService.process(JsonRpcHttpService.java:725)
        at org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcHttpService.lambda$handleJsonSingleRequest$13(JsonRpcHttpService.java:579)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:829)```

### PR DESCRIPTION
**Pull request by RatanRSur**
Signed-off-by: Ratan Rai Sur <ratan.r.sur@gmail.com>

## Fixed Issue(s)

fixes #2476 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java [hunk 1]
```diff
@@ -89,10 +89,11 @@ public static MinerDataResult createMinerDataResult(
...
                         .map(
                             receipt ->
-                                receipt
-                                    .getTransaction()
-                                    .calcEffectiveGas(receipt.getBaseFee())
-                                    .multiply(receipt.getGasUsed()))
+                                Wei.of(
+                                    receipt
+                                            .getTransaction()
+                                            .getEffectivePriorityFeePerGas(receipt.getBaseFee())
```
Comment (garyschulte) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java:95]: AFAIK - this should be effective gas, not effective priority fee

**Review by garyschulte — CHANGES_REQUESTED** [review]
I don't think we want to return effective priority fee, but rather effective gas price 🤔

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java (hunk 1 above)
Comment (RatanRSur) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java:95]: :man_facepalming: good point

**Review by garyschulte — APPROVED** [review]
clean and tight 🚢

**Pull request merged**